sdk/inc/azure/iot/az_iot_provisioning_client.h

```diff
@@ -453,11 +462,61 @@ typedef struct
 } az_iot_provisioning_client_payload_options;
 
 /**
+ * @brief Gets the default az_iot_provisioning_client_payload_options.
+ * @details Call this to obtain an initialized #az_iot_provisioning_client_payload_options structure
+ * that can be afterwards modified and passed to
+ * #az_iot_provisioning_client_register_get_request_payload.
+ *
+ * @return #az_iot_provisioning_client_payload_options.
+ */
+AZ_NODISCARD az_iot_provisioning_client_payload_options
+az_iot_provisioning_client_payload_options_default();
+
+/**
+ * @brief Builds the optional payload for a provisioning request.
+ * @remark Use this API to build an MQTT payload during device registration.
+ *         This call is optional for most scenarios. Some service
+ *         applications may require `custom_payload_property` specified during
+ *         registration to make additional decisions during provisioning time.
+ *         For example, if you need to register an IoT Plug and Play device you must
+ *         specify its model_id with this API via the `custom_payload_property`
+ *         `{"modelId":"your_model_id"}`.
+ *
+ * @param[in] client The #az_iot_provisioning_client to use for this call.
+ * @param[in] custom_payload_property __[nullable]__ Custom JSON to be added to this payload.
+ * Can be `NULL`.
+ * @param[in] options A reference to an #az_iot_provisioning_client_payload_options
+ * structure. Must be initialized first by calling
+ * az_iot_provisioning_client_payload_options_default() and then populating relevant options with
+ * your own values.
+ * @param[out] mqtt_payload A buffer with sufficient capacity to hold the MQTT payload.
+ * @param[in] mqtt_payload_size The size, in bytes of \p mqtt_payload.
+ * @param[out] out_mqtt_payload_length Contains the length, in bytes, written to \p mqtt_payload on
+ * success.
+ * @pre \p client must not be `NULL`.
+ * @pre \p mqtt_payload must not be `NULL`.
+ * @pre \p mqtt_payload_size must be greater than 0.
+ * @pre \p out_mqtt_payload_length must not be `NULL`.
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK The payload was created successfully.
+ * @retval #AZ_ERROR_NOT_ENOUGH_SPACE The buffer is too small.
+ */
+AZ_NODISCARD az_result az_iot_provisioning_client_register_get_request_payload(
+    az_iot_provisioning_client const* client,
+    az_span custom_payload_property,
+    az_iot_provisioning_client_payload_options const* options,
+    uint8_t* mqtt_payload,
+    size_t mqtt_payload_size,
+    size_t* out_mqtt_payload_length);
+
+/**
+ * @deprecated since 1.5.0-beta.1.
+ * @see az_iot_provisioning_client_register_get_request_payload
```

Comment:
This doesn't render the best in doxygen docs:
https://azuresdkdocs.blob.core.windows.net/$web/c/az_iot/1.5.0/deprecated.html

We should, at least update this to the following.
```suggestion
 * @deprecated since 1.5.0
 * @see az_iot_provisioning_client_register_get_request_payload
```

 I will leave it open for others to figure out other improvements :)

Reply:
This was the time the changes were deprecated. Feel free to submit a new PR to update.

Reply:
> This doesn't render the best in doxygen docs:

Can you be more specific about what doesn't render correctly? Looks acceptable to me:

![image](https://user-images.githubusercontent.com/8635911/211917453-02153890-0a0a-4ee6-8771-19fb9187a136.png)

https://azuresdkdocs.blob.core.windows.net/$web/c/az_iot/1.5.0/az__iot__provisioning__client_8h.html#a82fdc4e466e89adf377f1f5c15756d2f

Reply:
It looks good in the API list.

The link I shared leads to the `Deprecated List` page, which formats the function signature in a way that is difficult to parse/read.

It also don't understand what is meant by `Global`. Apparently there's a way to remove that page all together (https://stackoverflow.com/a/25065592), but maybe we should just keep it, as is.

Reply:
> This was the time the changes were deprecated. Feel free to submit a new PR to update.

Fixed in https://github.com/Azure/azure-sdk-for-c/pull/2480

Reply:
> This was the time the changes were deprecated. Feel free to submit a new PR to update.

Fixed in https://github.com/Azure/azure-sdk-for-c/pull/2480